server/src/main/java/org/elasticsearch/search/internal/ShardSearchRequest.java

```diff
@@ -357,6 +381,13 @@ protected final void innerWriteTo(StreamOutput out, boolean asKey) throws IOExce
                     + "] or greater."
             );
         }
+        if (out.getVersion().onOrAfter(Version.V_8_4_0)) {
+            out.writeBoolean(forceSyntheticSource);
+        } else {
+            if (forceSyntheticSource) {
+                throw new IllegalArgumentException("force_synthetic_source is not supported before 8.3.0");
```

Comment:
you mention 8.3 in the error message, but 8.4 in the conditional above. I suspect you did that on purpose but I am not following: is there a way to force synthetic source on 8.3 which does not support the flag?

Reply:
A mistake!

Reply:
Unbelievable, I could not believe my eyes hence I convinced myself you must have done it on purpose :)